internal/git/commands_test.go

```diff
@@ -315,6 +315,499 @@ func TestBackendCommands(t *testing.T) {
+				must.Eq(t, want, have)
+			})
+
+			t.Run("branch is deleted at the remote", func(t *testing.T) {
+				t.Parallel()
+				origin := testruntime.Create(t)
+				local := testruntime.Clone(origin.TestRunner, t.TempDir())
+				origin.CreateAndCheckoutFeatureBranch("branch", initial.Location())
+				origin.CreateCommit(testgit.Commit{
+					Branch:      "branch",
+					FileContent: "content",
+					FileName:    "file",
+					Message:     "commit on parent-only branch",
+				})
+				local.Fetch()
+				local.CheckoutBranch("branch")
+				origin.CheckoutBranch(initial)
+				err := origin.Git.DeleteLocalBranch(origin, "branch")
+				must.NoError(t, err)
+				local.Fetch()
+				initialCommits, err := local.Git.CommitsInBranch(local, initial, None[gitdomain.LocalBranchName]())
+				must.NoError(t, err)
+				branchCommits, err := local.Git.CommitsInBranch(local, "branch", None[gitdomain.LocalBranchName]())
+				must.NoError(t, err)
+				want := gitdomain.BranchesSnapshot{
+					Active: Some[gitdomain.LocalBranchName]("branch"),
+					Branches: gitdomain.BranchInfos{
+						gitdomain.BranchInfo{
+							LocalName:  Some[gitdomain.LocalBranchName]("branch"),
+							LocalSHA:   Some(branchCommits[0].SHA),
+							SyncStatus: gitdomain.SyncStatusDeletedAtRemote,
+							RemoteName: Some(gitdomain.NewRemoteBranchName("origin/branch")),
+							RemoteSHA:  None[gitdomain.SHA](),
+						},
+						gitdomain.BranchInfo{
+							LocalName:  Some(initial),
+							LocalSHA:   Some(initialCommits[1].SHA),
+							SyncStatus: gitdomain.SyncStatusUpToDate,
+							RemoteName: Some(gitdomain.NewRemoteBranchName("origin/initial")),
+							RemoteSHA:  Some(initialCommits[1].SHA),
+						},
+					},
+				}
+				have, _ := local.Git.BranchesSnapshot(local)
+				must.Eq(t, want, have)
+			})
+
+			t.Run("branch is active in another worktree", func(t *testing.T) {
+				t.Parallel()
+				runtime := testruntime.Create(t)
+				runtime.CreateBranch("branch", initial.BranchName())
+				worktreeDir := t.TempDir()
+				runtime.CreateWorktree(worktreeDir, "branch")
+				commits, err := runtime.Git.CommitsInBranch(runtime, initial, None[gitdomain.LocalBranchName]())
+				must.NoError(t, err)
+				want := gitdomain.BranchesSnapshot{
+					Active: Some[gitdomain.LocalBranchName]("initial"),
+					Branches: gitdomain.BranchInfos{
+						gitdomain.BranchInfo{
+							LocalName:  Some(gitdomain.NewLocalBranchName("branch")),
+							LocalSHA:   Some(commits[0].SHA),
+							SyncStatus: gitdomain.SyncStatusOtherWorktree,
+							RemoteName: None[gitdomain.RemoteBranchName](),
+							RemoteSHA:  None[gitdomain.SHA](),
+						},
+						gitdomain.BranchInfo{
+							LocalName:  Some(initial),
+							LocalSHA:   Some(commits[0].SHA),
+							SyncStatus: gitdomain.SyncStatusLocalOnly,
+							RemoteName: None[gitdomain.RemoteBranchName](),
+							RemoteSHA:  None[gitdomain.SHA](),
+						},
+					},
+				}
+				have, _ := runtime.Git.BranchesSnapshot(runtime)
```

Comment:
I prefer to verify the error returned by the function under test.

```suggestion
				have, err := runtime.Git.BranchesSnapshot(runtime)
				must.NoError(t, err)
```